.github/workflows/esp32.yml

```diff
@@ -0,0 +1,56 @@
+name: CI for ESP32
+
+on:
+  workflow_call:
+
+env:
+  CARGO_INCREMENTAL: 0
+  RUSTFLAGS: "-D warnings --cfg espidf_time32 "
```

Comment:
- I wouldn't expect the `time32` thing to be needed, as it is already in the top-level `.cargo/config.toml`.
- What is the `-D warnings` doing here?